### DISCUSSION TITLE
Bug - validation on update forms

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11512] [General]
Hi,

I believe i have encountered a bug when applying conditional validation to an update form.

We have a requirement for a user to optionally fill in a mobile number. If the field has a value i want to apply a regex function to ensure its formatted correctly, if there is no value the field is not mandatory.

To do this I am using the condition menu on the field so that it applies a must match regex, if the field is not empty:

![image](https://github.com/Budibase/budibase/assets/83434591/2f14ec10-8b86-405f-83c8-29fd97a8c630)

When doing this, and viewing an existing record. whenever i delete the data in the field it repopulates with the saved value as i click out of the field, so I cannot empty an existing record.

The form was created with a form block, then ejected but has no specific changes to how it loads data.

When I apply this conditional validation on the field in a create form, it works as expected.

See demo:
https://github.com/Budibase/budibase/assets/83434591/2e6c8505-3fed-4ac4-9f0e-ac1d9336e528

**Comment by melohagan** [comment]
Hey @benaunt 

I'm not able to reproduce this myself. 

Why are you using conditional UI in this case? You can just add validation to the form field directly?

![Screenshot 2023-08-14 at 10 09 06](https://github.com/Budibase/budibase/assets/101575380/58e6bdc4-38cb-4606-8e51-18b7b899dff1)

**Reply by 3DmotionMaker** [reply]
Hi, I experience something similar. 
Field validation also repopulates the field under certain conditions. It looks to me that if the value of the field is used in the validation, it repopulates. 

Below a reproducible scenario.
![image](https://github.com/user-attachments/assets/107ea848-7e33-4e9b-b5ce-88e2c14deff0)
![image](https://github.com/user-attachments/assets/ccc85a8d-22fb-44b3-96de-a3341000d29a)
